memory/iterator.go

```diff
@@ -0,0 +1,37 @@
+package memory
+
+import (
+	"io"
+
+	"gopkg.in/sqle/sqle.v0/sql"
+)
+
+type IteratorData interface {
+	Get(idx int) []interface{}
+	Length() int
+}
+
+type Iter struct {
+	idx  int
+	data IteratorData
+}
+
+func NewIter(data IteratorData) *Iter {
```

Comment:
I think it's better to make this type private and expose only the RowIter interface.